.editorconfig

```
@@ -0,0 +1,319 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+max_line_length = 120
+tab_width = 4
+ij_continuation_indent_size = 8
+ij_formatter_off_tag = @formatter:off
+ij_formatter_on_tag = @formatter:on
+ij_formatter_tags_enabled = true
+ij_smart_tabs = false
+ij_visual_guides =
+ij_wrap_on_typing = false
```

```suggestion
insert_final_newline = true
tab_width = 4
ij_continuation_indent_size = 8
```

indent size continuation I think is fine to keep as it'll actually properly matter
the other stuff like `formatter` won't matter as we don't ever use it nor will need to